File: src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionEval.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.adapters.JsonAdapter;
import co.nlighten.jsontransform.functions.common.ArgType;
import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;
import co.nlighten.jsontransform.functions.annotations.Aliases;
import co.nlighten.jsontransform.functions.annotations.Documentation;
import co.nlighten.jsontransform.functions.annotations.InputType;
import co.nlighten.jsontransform.functions.annotations.OutputType;

@Aliases("eval")
@Documentation("Evaluates the input and then transforms the context with the expression")
@InputType(ArgType.Any)
@OutputType(ArgType.Any)
public class TransformerFunctionEval<JE, JA extends Iterable<JE>, JO extends JE> extends TransformerFunction<JE, JA, JO> {

    public TransformerFunctionEval(JsonAdapter<JE, JA, JO> adapter) {
        super(adapter);
    }
    @Override
    public Object apply(FunctionContext<JE, JA, JO> context) {
        var eval = context.getJsonElement(null,true);
        return context.transform(eval, true);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionEval.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";

class TransformerFunctionEval extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const _eval = await context.getJsonElement(null);
    return await context.transform(context.getPath() + "/.", _eval, true);
  }
}

export default TransformerFunctionEval;
